qemu/tests/pvpanic_event_check.py

```diff
@@ -53,6 +56,18 @@ def run(test, params, env):
             "within the timeout period" % kdump_expect_status
         )
 
+    error_context.context("Check ISA Bridge in the guest", test.log.info)
+    o = session.cmd_output(check_ISA_cmd)
+    device_id = o.split()[0]
+    device_cmd = device_cmd % device_id
+    o = session.cmd_output(device_cmd)
+    if o.strip() == "1" and "i440fx" in machine_type or machine_type == "pc":
```

Comment:
Hi @heywji Here is an logic error.
`>>> a = 1
>>> m = "pc"
>>> if a == 1 and "i440fx" in m or m== "pc":
...     print(m)
... 
pc
>>> a = 2
>>> if a == 1 and "i440fx" in m or m== "pc":
...     print(m)
... 
pc
`

We should not print m when a=2. The correct code logic should be like this:

`>>> if a == 1 and ("i440fx" in m or m== "pc"):
...     print(m)
...`

```suggestion
    if o.strip() == "1" and ("i440fx" in machine_type or machine_type == "pc"):
```

Reply:
Thanks for pointing that out, let me recommit and rerun the test result.

Reply:
Test Result:
```
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
1-Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads Finshed 2024-04-01 12:39:07 2024-04-01 12:52:04 PASS 776.783087
2-Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35 Finshed 2024-04-01 12:52:04 2024-04-01 13:05:20 PASS 795.887213
3-Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check Finshed 2024-04-01 13:05:21 2024-04-01 13:07:21 FAIL 120.170622
4-Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-04-01 13:07:21 2024-04-01 13:09:17 PASS 115.666028
Summary:
Finshed=4, PASS=3, FAIL=1
```